### PR TITLE
Implement schema methods in the HttpManagementProxy (fixes #1344)

### LINKED ISSUE #1344: Implement schema methods in the HttpManagementProxy
[Project board link](https://github.com/orgs/k8ssandra/projects/8/views/1?pane=issue&itemId=35571825)

## Definition of Done
- [x] the `getKeyspaces()` method is implemented in the HttpCassandraManagementProxy class
- [x] the `getColumnFamilyName()` method is implemented in the HttpCassandraManagementProxy class

### PR DESCRIPTION
**Pull request by olim7t**
Fixes #1344 

Prerequisites:
- [x] merge k8ssandra/management-api-for-apache-cassandra#352
- [x] update the `datastax-mgmtapi-client-openapi` version in `src/server/pom.xml`.

### CODE REVIEW

File: src/server/src/main/java/io/cassandrareaper/management/http/HttpCassandraManagementProxy.java [hunk 1]
```diff
@@ -159,12 +161,27 @@ public String getClusterName() {
 
   @Override
   public List<String> getKeyspaces() {
```
Comment (Miles-Garnsey) [inline src/server/src/main/java/io/cassandrareaper/management/http/HttpCassandraManagementProxy.java:163]: Issue: Can we please have a unit test for this method? Maybe something that mocks `apiClient.listKeyspaces()` and then just confirms that it was called?

File: src/server/src/main/java/io/cassandrareaper/management/http/HttpCassandraManagementProxy.java [hunk 2]
```diff
@@ -159,12 +161,27 @@ public String getClusterName() {
...
+    try {
+      return apiClient.listKeyspaces("");
+    } catch (ApiException ae) {
+      LOG.error("Failed to list keyspaces", ae);
+      return Collections.emptyList();
+    }
   }
 
   @Override
   public Set<Table> getTablesForKeyspace(String keyspace) throws ReaperException {
```
Comment (Miles-Garnsey) [inline src/server/src/main/java/io/cassandrareaper/management/http/HttpCassandraManagementProxy.java:173]: Issue: Would be good to see a unit test for this too please :) Similar procedure to the other one, mock the call to the apiClient method and confirm that the processing in the rest of the method does what it should.

**Review by Miles-Garnsey — CHANGES_REQUESTED** [review]
Sorry this has taken me a little while to get to, I thought someone else was reviewing it!

Just requesting changes to add unit tests, let me know if you have any concerns.

File: src/server/src/main/java/io/cassandrareaper/management/http/HttpCassandraManagementProxy.java (hunk 1 above)
Comment (adejanovski) [reply src/server/src/main/java/io/cassandrareaper/management/http/HttpCassandraManagementProxy.java:163]: @Miles-Garnsey , is this a **todo:** or a **suggestion:** ?

Comment (Miles-Garnsey) [reply src/server/src/main/java/io/cassandrareaper/management/http/HttpCassandraManagementProxy.java:163]: Sorry, an issue I think.

File: src/server/src/main/java/io/cassandrareaper/management/http/HttpCassandraManagementProxy.java (hunk 2 above)
Comment (olim7t) [reply src/server/src/main/java/io/cassandrareaper/management/http/HttpCassandraManagementProxy.java:173]: I added both unit tests as requested.

**Review by Miles-Garnsey — APPROVED** [review]
This looks like it is all passing (there is one still running but I'm confident), those tests look great too. Thanks for adding a generic method to create the mocked proxy.

**Comment by Miles-Garnsey** [comment]
I am merging this because all tests are green, it is approved, and I want the `mockProxy()` method for my own tests.

**Pull request merged**